src/main/java/jchess/ruleengine/KingRulesProcessor.java

```diff
@@ -0,0 +1,196 @@
+				
+				continue;
+			}
+			boolean bSameName = pcNextField.getName().equals("Rook" + oPiece.getFamily());
+			// otherwise If he's a rook and he has never moved
+			if(bSameName && pcNextField.getPositionHistoryCount() == 0) 
+			{		
+				IRule oRule = m_oBoardFactory.createRule();		
+				oRule.getRuleData().setRuleType(RuleType.CUSTOM);
+				oRule.getRuleData().setDirection(Direction.EDGE);
+				oRule.getRuleData().setMaxRecurrenceCount(Integer.MAX_VALUE);
+				oRule.getRuleData().setFile(fDirection);
+				oRule.getRuleData().setRank(Rank.SAME);
+				oRule.getRuleData().setFamily(Family.IGNORE);
+				oRule.getRuleData().setManoeuvreStrategy(Manoeuvre.FILE_AND_RANK);
+				oRule.getRuleData().setCustomName("MOVE[KING_CASTLING]");
+				((IRuleAgent)oRule).reset();
+				
+				for (IPositionAgent moveCandidateEntry : pathToRook) {
+					if(isFieldEndangered(moveCandidateEntry, oRuleProcessor, oBoard, oCurrentPlayer)) {
+						return;
+					}
+					// if field is endangered then quit the function
+				}
+				
+				// if we got this far the path is safe and we can add the move candidate to the map of possible move candidates.
+				IMoveCandidate oCastlingMoveKingCandidate = new MoveCandidate((IRuleAgent)oRule, oPiece, oPiece.getPosition(), possibleKingTarget);
+				IMoveCandidate oCastlingMoveRookCandidate = new MoveCandidate(null, posNextField.getPiece(), posNextField, possibleRookTarget);
+				
+				oCastlingMoveKingCandidate.addSecondaryMove(oCastlingMoveRookCandidate);
+				mpCandidateMovePositions.put(possibleKingTarget.getName(), oCastlingMoveKingCandidate);
+				
+				break;				
+			
+			}
+			else {
+				// no rook was found.
+				break;
+			}
+		}
+	}
+	
+	/**
+	 * This method executes the castling move.
+	 * 
+	 * @param IBoardAgent
+	 * @param IMoveCandidate
+	 * @return IBoardActivity
+	 */
+	public static IBoardActivity tryExecuteCastlingRule(IBoardAgent oBoard, IMoveCandidate oMoveCandidate) {
+		// move the king
+		IBoardActivity oActivity = new BoardActivity(oMoveCandidate);
+		
+		IPieceAgent oPieceLinkedToCurrentPosition = oMoveCandidate.getSourcePosition().getPiece();
+		IPositionAgent oCurrentPosition = oMoveCandidate.getSourcePosition();
+		IPositionAgent oNewPosition = oMoveCandidate.getCandidatePosition();
+		
+		oPieceLinkedToCurrentPosition.setPosition(oNewPosition);
+		oNewPosition.setPiece(oPieceLinkedToCurrentPosition);
+		oCurrentPosition.setPiece(null);
+		
+		oPieceLinkedToCurrentPosition.enqueuePositionHistory(oCurrentPosition);
+		
+		oActivity.addPriorMoveEntry(oCurrentPosition, oPieceLinkedToCurrentPosition);
+		oActivity.addPriorMoveEntry(oNewPosition, null);
+		oActivity.addPostMoveEntry(oCurrentPosition, null);
+		oActivity.addPostMoveEntry(oNewPosition, oPieceLinkedToCurrentPosition);		
+		
+		// move the rook
+		for (IMoveCandidate secondaryMoveCandidate : oMoveCandidate.getSecondaryMoves()) {
+			secondaryMoveCandidate.getPieceToMove().setPosition(secondaryMoveCandidate.getCandidatePosition());
```

Comment:
if you make temporary variables for piece, and positions for secondary move candidates here it would be easy to address them later in the function.

Reply:
done.